src/main/java/io/cdap/plugin/servicenow/sink/service/ServiceNowSinkAPIRequestImpl.java

```diff
@@ -0,0 +1,251 @@
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import io.cdap.cdap.api.retry.RetryableException;
+import io.cdap.plugin.servicenow.apiclient.ServiceNowTableAPIClientImpl;
+import io.cdap.plugin.servicenow.apiclient.ServiceNowTableAPIRequestBuilder;
+import io.cdap.plugin.servicenow.restapi.RestAPIResponse;
+import io.cdap.plugin.servicenow.sink.ServiceNowSinkConfig;
+import io.cdap.plugin.servicenow.sink.model.RestRequest;
+import io.cdap.plugin.servicenow.sink.model.ServiceNowBatchRequest;
+import io.cdap.plugin.servicenow.util.ServiceNowConstants;
+import org.apache.http.Header;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpStatus;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.message.BasicHeader;
+import org.apache.oltu.oauth2.common.exception.OAuthProblemException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import javax.ws.rs.core.MediaType;
+
+/**
+ * Implementation class for ServiceNow Batch Rest API.
+ */
+public class ServiceNowSinkAPIRequestImpl {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ServiceNowSinkAPIRequestImpl.class);
+
+  private static Integer counter = 1;
+  private static Integer batchRequestIdCounter = 1;
+  private final ServiceNowSinkConfig config;
+  private final ServiceNowTableAPIClientImpl restApi;
+  private final Gson gson = new Gson();
+  private final JsonParser jsonParser = new JsonParser();
+  private boolean isCreated;
+
+
+  public ServiceNowSinkAPIRequestImpl(ServiceNowSinkConfig conf) {
+    this.config = conf;
+    restApi = new ServiceNowTableAPIClientImpl(config);
+  }
+
+  public RestRequest getRestRequest(JsonObject jsonObject) {
+    JsonElement jsonElement = jsonObject.get(ServiceNowConstants.SYS_ID);
+    String sysId = null;
+
+    if (jsonElement == null) {
+      if (config.getOperation().equals(ServiceNowConstants.UPDATE_OPERATION)) {
+        throw new RuntimeException("No sys_id found in the record to be updated");
+      }
+    } else {
+      if (config.getOperation().equals(ServiceNowConstants.INSERT_OPERATION)) {
+        jsonObject.remove(ServiceNowConstants.SYS_ID);
+      }
+      sysId = jsonElement.getAsString();
+    }
+
+    String data = jsonObject.toString();
+    String encodedData = Base64.getEncoder().encodeToString(data.getBytes(StandardCharsets.UTF_8));
+
+    List<Header> headers = new ArrayList<>();
+    Header contentTypeHeader = new BasicHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON);
```

Comment:
write a separate method getHeaders for generic headers and reuse it.

Reply:
Created a private `getHeaders()` method for forming the list of headers 659b6bd2dc503d409443ef1b43804566a4e25be0